WordPress/Classes/ViewRelated/Notifications/Milestone Notifications/ConfettiView.swift

```diff
@@ -0,0 +1,193 @@
+            transition.type = .fade
+            transition.duration = duration * 0.5
+            transition.timingFunction = CAMediaTimingFunction(name: .easeOut)
+            transition.setValue(layer, forKey: Constants.animationLayerKey)
+            transition.isRemovedOnCompletion = false
+
+            layer.add(transition, forKey: nil)
+            layer.opacity = 0
+        }
+
+        layer.add(animation, forKey: nil)
+        CATransaction.commit()
+    }
+
+    // MARK: - Private: ParticleEmitterLayer
+    private class ParticleEmitterLayer: CAEmitterLayer {
+        init(with particles: [Particle], config: EmitterConfig) {
+            super.init()
+
+            needsDisplayOnBoundsChange = true
+            emitterCells = particles.map { ParticleCell(with: $0, config: config) }
+        }
+
+        override func layoutSublayers() {
+            super.layoutSublayers()
+
+            emitterMode = .outline
+            emitterShape = .line
+            emitterSize = CGSize(width: bounds.width, height: 1.0)
+            emitterPosition = CGPoint(x: bounds.midX, y: 0)
+        }
+
+        override init(layer: Any) {
+            super.init(layer: layer)
+        }
+
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        private class ParticleCell: CAEmitterCell {
+            init(with particle: Particle, config: EmitterConfig) {
+                super.init()
+
+                contents = particle.tintedImage().cgImage
+                birthRate = config.birthRate
+                lifetime = config.lifetime
+                scale = config.scale
+                scaleRange = config.scaleRange
+                beginTime = CACurrentMediaTime()
+                velocity = CGFloat(birthRate * lifetime)
+                velocityRange = velocity * 0.5
+                emissionLongitude = .pi
+                emissionRange = .pi / 4
+                spinRange = .pi * 8
+            }
+
+            required init?(coder: NSCoder) {
+                fatalError("init(coder:) has not been implemented")
+            }
+        }
+    }
+
+    private struct Constants {
+        static let animationLayerKey = "org.wordpress.confetti"
+    }
+}
+
+// MARK: - Animation Delegate
+extension ConfettiView: CAAnimationDelegate {
+    func animationDidStop(_ anim: CAAnimation, finished flag: Bool) {
+        guard let layer = anim.value(forKey: Constants.animationLayerKey) as? ParticleEmitterLayer else {
+            return
+        }
+
+        layer.removeAllAnimations()
+        layer.removeFromSuperlayer()
+    }
+}
+
+
+extension ConfettiView {
+
+    func emitConfetti() {
+        // Images
+        let star = UIImage(named: "confetti-star")!
+        let circle = UIImage(named: "confetti-circle")!
+        let hotdog = UIImage(named: "confetti-hotdog")!
+
+        // Colors
+        let purple = UIColor(red: 0.75, green: 0.35, blue: 0.95, alpha: 1.00)
```

Comment:
Should we translate these to muriel colors?

Reply:
I don't think these colors are in the muriel set. cc @mbshakti ?